modules/imgproc/test/test_drawing.cpp

```diff
@@ -245,6 +248,46 @@ int CV_DrawingTest_CPP::checkLineIterator( Mat& img )
     return 0;
 }
 
+int CV_DrawingTest_CPP::checkLineVirtualIterator(  )
+{
+    RNG randomGenerator(1);
+    for (size_t test = 0; test < 10000; ++test)
+    {
+        int width = randomGenerator.uniform(0, 512+1);
+        int height = randomGenerator.uniform(0, 512+1);
+        int x1 = randomGenerator.uniform(-512, 1024+1);
+        int y1 = randomGenerator.uniform(-512, 1024+1);
+        int x2 = randomGenerator.uniform(-512, 1024+1);
+        int y2 = randomGenerator.uniform(-512, 1024+1);
+        int channels = randomGenerator.uniform(1, 3+1);
+        Mat m(cv::Size(width, height), CV_MAKETYPE(8U, channels));
+        LineIterator it( m, Point(x1,y1), Point(x2, y2) );
+        LineVirtualIterator vit(Rect(0, 0, width, height), Point(x1,y1), Point(x2, y2) );
```

Comment:
This checks only the case when `currentPosOffset` is (0, 0), so can be reduced to constructor with the `Size` as the first argument. Consider to add case for non-zero offset too.